worker-queue-rabbit/src/main/java/com/github/workerframework/queues/rabbit/WorkerQueueConsumerImpl.java

```diff
@@ -180,14 +227,17 @@ private void processReject(long id, boolean requeue)
      *
      * @param delivery the redelivered message
      */
-    private void republishClassicRedelivery(final Delivery delivery, final int retries) {
+    private void republishClassicRedelivery(final Delivery delivery, final int retries, final Optional<String> dehydratedMessageId) {
 
         final RabbitTaskInformation taskInformation = 
                 new RabbitTaskInformation(String.valueOf(delivery.getEnvelope().getDeliveryTag()));
         LOG.debug("Received redelivered message with id {}, retry count {}, retry limit {}, republishing to retry queue",
                 delivery.getEnvelope().getDeliveryTag(), retryLimit, retries + 1);
         final Map<String, Object> headers = new HashMap<>();
         headers.put(RabbitHeaders.RABBIT_HEADER_CAF_WORKER_RETRY, String.valueOf(retries + 1));
+        if (dehydratedMessageId.isPresent()) {
+            headers.put(RABBIT_HEADER_CAF_DEHYDRATION_ID, dehydratedMessageId.get());
+        }
```

Comment:
For a poison message, using classic queues, won't the dehydratedMessageId be deleted when the delivered message is ack'd?

Also the delivery.getMessageData() will return an empty byte array for a dehydrated message?

Reply:
If we call republishClassicRedelivery, the incoming dehydratedMessageId is not set in the TaskInformation so it will not be deleted.   Yes the messageData will be empty if its a dehydrated message, but it will be re-hydrated on re-delivery.